octree_web_viewer/src/state.rs

```diff
@@ -0,0 +1,101 @@
+use crate::backend_error::PointsViewerError;
+use point_viewer::octree;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+
+/// path information for the octrees
+#[derive(Clone)]
+pub struct OctreeKeyParams {
+    /// Location prefix, including
+    prefix: String,
+    /// Tree ID
+    suffix: String,
+}
+
+impl OctreeKeyParams {
+    pub fn get_octree_address(&self, octree_key: &String) -> Result<String, PointsViewerError> {
+        let mut join_prefix = "";
+        let mut join_suffix = "";
+
+        if !self.prefix.ends_with("/") {
+            join_prefix = "/";
+        }
+        if !self.suffix.starts_with("/") {
+            join_suffix = "/";
+        }
+
+        Ok(format!(
+            "{}{}{}{}{}",
+            self.prefix, join_prefix, octree_key, join_suffix, self.suffix
```

Comment:
Would it make sense to use `PathBuf`/`Path` for the components and `join` them?

Reply:
indeed, done!